internal/lock.go

```diff
@@ -150,11 +150,22 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string)
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-	for _, r := range filteredResources {
+
+	var statusLine *StatusLine
+	if bar {
+		statusLine = NewStatusLine(filteredResources)
+	}
+
+	for i, r := range filteredResources {
 		resource := r
 		go func() {
+
 			err := resource.Download(dir, mode, ctx)
 			errorCh <- err
+
+			if bar {
```

Comment:
nit: Instead of checking bar, maybe check if statusLine is empty. This is cleaner in the sense that it will let you potentially refactor the code related to `bar` without having to change other things.